tools/ci_build/github/azure-pipelines/win-ci-on-device-training.yml

```diff
@@ -0,0 +1,21 @@
+parameters:
+- name: RunOnnxRuntimeTests
```

Comment:
I suggest it'd better to split the workflow into 2 stage, building & testing in the next PRs.
some stages could be skipped in pipeline portal and different stages could run in different pools.

Reply:
I didn't quite follow the suggestion.
This file is just a modification of an existing file `tools/ci_build/github/azure-pipelines/win-ci-pipeline.yml` (Windows CPU CI Pipeline for ort) for the on-device-training scenario (addition of the `--enable_training_on_device` flag in the build args). As such, if you look at the template it invokes: `templates/win-cpu-ci.yml`, the build happens in a different step compared to the run tests step.
You can view the pipeline run [here](https://dev.azure.com/onnxruntime/onnxruntime/_build/results?buildId=717535&view=logs&j=78245837-8ee8-5268-5ba6-4858d2b4c8d6&t=172d9be8-96dc-5dc5-4044-7f56c5e4e79c).

Maybe I missed what you are trying to suggest.

Reply:
win-cpu-ci.yml is not updated to support it yet. I think the concern can be address in a new PR, for all Windows GPU build pipelines.

Reply:
@baijumeswani, I think he means we should use cpu machines for building code, gpu machine for testing, given the number of GPU machines is limited.

Reply:
Thanks. may we pick up this work in a follow up PR?